cancellation.md

```diff
@@ -28,15 +37,51 @@ __R1__
 
 # Introduction
 
-One of the basis operations for any async function is cancellation. In this paper we explore the uses of cancellation to determine how to represent the result of a cancelled async function (the mechanism to signal a request for cancellation is covered by `stop_source` in C++20). In this paper a cancelled result is described as an instance of serendipitous-success ([Credits] go to Lisa Lippincott for coining this term).
+One of the basis operations for any async function is cancellation. In this paper we explore the uses of cancellation to determine how to represent the result of a cancelled async function (the mechanism to signal a request for cancellation is covered by `stop_source` in C++20). In this paper a cancelled result is described as an instance of serendipitous-success (Credit to Lisa Lippincott for coining this term).
+
+## cancellation is not just for async functions
+
+In writing this paper it became clear that serendipitous-success is essential for all C++ functions. C++ already supports serendipitous-success within blocks using `goto` and the more constrained `break` and `continue` to skip over code without throwing an exception. However these do not compose in a way that allows a stack of functions to be unwound without throwing an exception. [@P0709R4] _Zero-overhead deterministic exceptions: Throwing values_ describes mechanisms that could possibly be extended to enable serendipitous-success for C++ functions. [@P1095R0] _Zero overhead deterministic failure - A unified mechanism for C and C++_ may allow serendipitous-success to be supported for C functions as well.
+
+## Dual of the Strategy pattern
+
+An effort is cancelled when some __higher-level__ goal __has already been satisfied__, rendering the current effort unnecessary, and possibly wasteful.
+
+This pattern is dual to the strategy pattern where we use exceptions. In the strategy pattern, we abort a strategy when we discover that a __low-level__ goal __cannot be satisfied__, rendering the current strategy impossible.
+
+Both patterns share the idea of early exit from an operation, and the thinking that has gone into the early exit model for exceptions also applies to cancellation.  In particular:
```

Comment:
I changed the phrase 'abnormal exit' to 'early exit'. Does that retain the meaning you intended?